lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -1877,6 +1882,50 @@ def test_remove_cell_measure(self):
                          [[self.b_cell_measure, (0, 1)]])
 
 
+class Test_share_data(tests.IrisTest):
+    def setter_lazy_data(self):
```

Comment:
I don't believe this will be run: test methods needs to be prefixed with `test_` for unittest to pick them up.